lib/abstract_format.ml

```diff
@@ -44,6 +44,7 @@ and literal_t =
   | LitString of {line: line_t; str: string}
 
 and pattern_t =
+  | PatBitstr of {line: line_t; elements: (pattern_t * expr_t option * (type_spec_t list) option) list}
```

Comment:
nits: `elements` seems a bit complicated. How about define a type like `pattern_bitstring_elems_t`?

Reply:
Thank you for your review!
I'll fix it.

Reply:
I added `{expr|pattern|guard_test}_bin_element_t` because this is called as `bin_element` in the document of absform.
Is it OK?

Reply:
:+1:

Reply:
Thank you! I'll merge this.